bgpd/bgp_debug.c

```diff
@@ -2522,17 +2551,28 @@ static bool bgp_debug_per_peer(char *host, unsigned long term_bgp_debug_type,
 		if (!per_peer_list || list_isempty(per_peer_list))
 			return true;
 
-		else {
-			if (!host)
-				return false;
+		if (!host)
+			return false;
 
-			for (ALL_LIST_ELEMENTS(per_peer_list, node, nnode,
-					       filter))
-				if (strcmp(filter->host, host) == 0)
-					return true;
+		for (ALL_LIST_ELEMENTS(per_peer_list, node, nnode, filter))
+			if (strmatch(filter->host, host) &&
+			    filter->plist_name && p) {
+				struct prefix_list *plist;
+				afi_t afi = family2afi(p->family);
 
-			return false;
-		}
+				plist = prefix_list_lookup(afi,
```

Comment:
Should move the cost of this lookup to when the configuration is made, i.e. instead of storing only `filter->plist_name` we should have `char *plist_name` + `struct prefix_list *plist_v4` + `struct prefix_list *plist_v6` and update the pointers in the appropriate prefix list notification hooks.

(Again, this code can be called a *lot*, it should be fast.)

Reply:
Will do.

Reply:
Fixed.